python/src/spark_rapids_ml/umap.py

```diff
@@ -894,6 +910,9 @@ def __init__(
         labelCol: Optional[str] = None,
         outputCol: Optional[str] = None,
         num_workers: Optional[int] = None,
+        enable_sparse_data_optim: Optional[
+            bool
+        ] = None,  # TBD: reuses 'vector -> sparse csr' conversion code, but will enable sparse data if first row is sparse for any metric. maybe set False by default, and None for 'jaccard'?
```

Comment:
It's only jaccard that [requires](https://github.com/rapidsai/cuml/blob/branch-24.10/python/cuml/cuml/manifold/umap_utils.pyx#L157) sparse inputs, but all other metrics can be used for both. I think None is the correct default here to allow sparse inputs for any metric.

Reply:
Got you. For jaccard, maybe we can add to docsting to explain only sparse is supported. Or it will be even better if we can check if inputs are sparse before calling cuml fit.